components/device-mgt/org.wso2.carbon.device.mgt.core/src/main/java/org/wso2/carbon/device/mgt/core/geo/service/GeoLocationProviderServiceImpl.java

```diff
@@ -349,6 +607,17 @@ private String getExecutionPlanName(String alertType, String queryName, String d
         }
     }
 
+    private String getExecutionPlanName(String alertType, String queryName) {
+        if ("Traffic".equals(alertType)) {
+            return "Geo-ExecutionPlan-Traffic_" + queryName + "_alert";
+        } else {
+            if (alertType.equals("Speed")) {
```

Comment:
instead use `if("Speed".equals(alertType))` to avoid possible NPE

Reply:
Fixed c653051